tenseal/utils.h

```diff
@@ -33,6 +33,47 @@ parameters.
 */
 shared_ptr<seal::SEALContext> create_context(EncryptionParameters parms);
 
+/*
+Replicate the current vector as many times to fill `final_size` elements.
+*/
+template <typename T>
+void replicate_vector(vector<T>& vec, size_t final_size) {
+    size_t init_size = vec.size();
+    vec.reserve(final_size);
+    for (size_t i = 0; i < final_size - init_size; i++) {
+        vec.push_back(vec[i % init_size]);
+    }
+}
+
+/*
+Returns the k-th diagonal of a matrix. Positive values of k represent upper
+diagonals while negative values represent lower diagonal with 0 being the main
+diagonal. It's important to note that diagonals here are of the size of the
+matrix, we rotate as we reach the boundaries of the matrix.
```

Comment:
it would actually be the intended output, I think I just didn't explain it well, I couldn't figure out a mathematical name for what I'm doing. It's like going all over the k-th diagonal, then if we reach the bottom we start from the top again, if reach the right we start from the left again, until getting out either the whole matrix, or a replicated diagonal, we could stop at the latter because it's just replicating the same diagonal, but it doesn't hurt because we will be replicating it anyway.

The whole point of this function is to align with encrypted vectors in a way to allow a chained of matrix multiplication later on.